### PR TITLE
Add Marketing sharing feature

### PR DESCRIPTION
**Pull request by lachlanjc**

### CODE REVIEW

**Comment by maxwofford** [comment]
Rendering logs a BUNCH of warnings now & will make it hard to debug in the future. Please fix this:

```
WARNING on line 33, column 10 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `red' in interpolation here.
It may end up represented as #ff0000, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "red").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 33, column 10 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `orange' in interpolation here.
It may end up represented as #ffa500, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "orange").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 33, column 10 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `yellow' in interpolation here.
It may end up represented as #ffff00, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "yellow").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 33, column 10 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `green' in interpolation here.
It may end up represented as #008000, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "green").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 33, column 10 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `blue' in interpolation here.
It may end up represented as #0000ff, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "blue").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 55, column 4 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `black' in interpolation here.
It may end up represented as #000000, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "black").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 59, column 7 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `black' in interpolation here.
It may end up represented as #000000, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "black").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 63, column 7 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `black' in interpolation here.
It may end up represented as #000000, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "black").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 55, column 4 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `snow' in interpolation here.
It may end up represented as #fffafa, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "snow").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 59, column 7 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `snow' in interpolation here.
It may end up represented as #fffafa, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "snow").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 63, column 7 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `snow' in interpolation here.
It may end up represented as #fffafa, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "snow").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 55, column 4 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `white' in interpolation here.
It may end up represented as #ffffff, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "white").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 59, column 7 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `white' in interpolation here.
It may end up represented as #ffffff, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "white").
If you really want to use the color value here, use `"" + $name'.

WARNING on line 63, column 7 of /usr/src/app/app/assets/stylesheets/_utilities.scss:
You probably don't mean to use the color value `white' in interpolation here.
It may end up represented as #ffffff, which will likely produce invalid CSS.
Always quote color names when using them as strings (for example, "white").
If you really want to use the color value here, use `"" + $name'.

  Rendered layouts/_nav.html.erb (14.8ms)
Completed 200 OK in 4547ms (Views: 4265.2ms | ActiveRecord: 19.7ms)
```

**Comment by maxwofford** [comment]
As an admin without a club, I get auth errors on the marketing page

screenshot 2018-11-19 22 36 57

File: app/models/user.rb [hunk 1]
```diff
@@ -31,6 +33,10 @@ def leader?
     self.leader.present?
   end
 
+  def club
+    clubs.first
```
Comment (maxwofford) [inline app/models/user.rb:37]: Why do this?

File: config/storage.yml [hunk 2]
```diff
@@ -6,29 +6,9 @@ local:
...
-# mirror:
-#   service: Mirror
-#   primary: local
-#   mirrors: [ amazon, google, microsoft ]
+amazon:
+  service: S3
+  access_key_id: <%= Rails.application.credentials.dig(:aws, :access_key_id) %>
+  secret_access_key: <%= Rails.application.credentials.dig(:aws, :secret_access_key) %>
+  region: <%= Rails.application.credentials.dig(:aws, :region) %>
+  bucket: <%= Rails.application.credentials.dig(:aws, :bucket) %>
```
Comment (maxwofford) [inline config/storage.yml:14]: Should the credentials file be updated to account for these new values? Don't see it in the PR.

File: app/models/user.rb (hunk 1 above)
Comment (maxwofford) [reply app/models/user.rb:37]: * specifically, this gives confusing implications (`user.club` tells me a user `has_one` club). What if you name it `first_club` or something like that?

File: config/storage.yml (hunk 2 above)
Comment (lachlanjc) [reply config/storage.yml:14]: Yes, but I don’t have those credentials :) Can you or @zachlatta set them?

**Comment by lachlanjc** [comment]
Once those credentials are in, this is ready!

File: config/storage.yml (hunk 2 above)
Comment (maxwofford) [reply config/storage.yml:14]: Moving this to Slack DM

**Pull request merged**